src/tasks/MonoTargetsTasks/ILStrip/AssemblyStripper/AssemblyStripper.cs

```diff
@@ -12,6 +12,16 @@
 
 namespace AssemblyStripper
 {
+    class CustomAttrRowComparer : IComparer
+    {
+        public int Compare(object left, object right)
+        {
+            CustomAttributeRow row_left = (CustomAttributeRow)left;
+            CustomAttributeRow row_right = (CustomAttributeRow)right;
+            return row_left.Parent.RID.CompareTo(row_right.Parent.RID);
```

Comment:
This is subtly wrong.  the problem is that `RID` on a cecil metadata token masks out the token type.

We actually have to reconstruct the custom attribute coded-index.
something like:

```csharp
   var leftParentCodedIdx = Utilities.CompressMetadataToken(CodedIndex.HasCustomAttribute, row_left.Parent.RID);
   var rightParentCodedIdx = Utilities.CompressMetadataToken(CodedIndex.HasCustomAttribute, row_right.Parent.RID);
   return leftParentCodedIdex.CompareTo(rightParentCodedIndex);
```

Reply:
This fixed the issue I was running into on https://github.com/dotnet/runtime/pull/88167